### PR TITLE
quantity form and addSelected() item connect

### PR DESCRIPTION
**Pull request by andrisLaduzans**
added number form in ./Product + connected addSelectedItem function to ./Product which is taking product:object with count:number as parameters. 
TODO is updated - next want to implement Formik library for the quantity input processing and validation. i know i can do it without Formik, but i want to learn how to use it since sighn up forms and forms in general are such a huge part of internet.

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -43,8 +21,9 @@ const ITEMS: IShopItems = [
 
 const App: React.FC = () => {
   const [selectedItems, setSelectedItems] = useState<ISelectedItem[]>([]);
-  function addSelectedItem(id: number):void{
-    alert('fuuuck!!!!')
```
Comment (andrisLaduzans) [inline src/App.tsx]: oops, sory about that

File: src/App.tsx [hunk 2]
```diff
@@ -43,8 +21,9 @@ const ITEMS: IShopItems = [
...
+  
+  function addSelectedItem(product:IShopItem, count:number):void{
+    console.log(product,count);
   }
```
Comment (andrisLaduzans) [inline src/App.tsx]: program stops here for now console logs the item which as been "added to basket"  and how many.

File: src/Product.tsx [hunk 3]
```diff
@@ -11,58 +10,59 @@ export interface IShopItem {
...
-    } else {
-      setCount(count + 1);
-    }
-  };
+  function handleSubmit(event: any) {
+    selectProduct(product, count);
+    event.preventDefault();
+  }
+  function handleChange(event: any) {
+    setCount(event.target.count);    //throws warning here controlled -> uncontrolled?
```
Comment (andrisLaduzans) [inline src/Product.tsx]: when run in browser it thows warning :Warning: A component is changing a controlled input of type number to be uncontrolled. Input elements should not switch from controlled to uncontrolled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. 
As soon as i change the number values... i red https://reactjs.org/docs/forms.html#controlled-components but still could not figure out what's the issue here.

File: src/Product.tsx [hunk 4]
```diff
@@ -11,58 +10,59 @@ export interface IShopItem {
...
+        <form onSubmit={handleSubmit}>
+          <input
+            type="number"
+            value={count}
+            min={1}
+            max={product.quantity}
+            onChange={handleChange}
+          />
+          <input className="submitBtn" type="submit" value="Add To Basket" />
+        </form>
```
Comment (andrisLaduzans) [inline src/Product.tsx]: planning to add Formik + yup form validation here....sometime in the future.

File: src/Product.tsx (hunk 3 above)
Comment (sandris-) [reply src/Product.tsx]: that is a common input problem, ask at the lecture for help

**Pull request merged**